src/QsFmt/Formatter/Formatter.fs

```diff
@@ -58,9 +59,21 @@ let format source =
 
 [<CompiledName "Update">]
 let update source =
-    parse source
-    |> Result.map (curry qubitBindingUpdate.Document () >> curry unitUpdate.Document () >> printer.Document)
+    let updateDocument document =
 
+        let updatedDocument =
+            document
+            |> curry qubitBindingUpdate.Document ()
+            |> curry unitUpdate.Document ()
+            |> curry arraySyntaxUpdate.Document ()
+
+        let warningList = updatedDocument |> updateChecker
+        let printedDocument = updatedDocument |> printer.Document
+
+        // For now, let's just combine the warnings to the end of the document
```

Comment:
Also, would it make sense to output these to stderr instead of just adding to the doc? I'm not sure how that would work with IDE integration and it might make automated testing trickier, but just thinking out loud here.

Reply:
My thinking here is that output in general is going to be reworked when the File I/O task for the tool is addressed. At that point, the formatted/updated content doesn't output to the console anymore, and the warnings would still output to the console. I'm not sure what effect outputting to the stderr has, but I can look into that.

Reply:
Printing to the stderr will make the warnings print before the updated content on the cmd output, but I guess that is fine since that is temporary until the file I/O is addressed anyway.

Reply:
Sadly that doesn't work well with the way testing is set up. I think the unit testing system that this tool has is going to need to be reworked sometime soon. I don't want to have to do that in this PR though, and this issue is ultimately an I/O one, so my thinking is to leave it as it is for this PR and address this as well as the testing framework in later PRs.

Reply:
Yeah, that makes sense. A testing overhaul should be a separate PR.